components/public-api/gitpod/v1/envvar.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service EnvironmentVariableService {
+  // ListEnvironmentVariables returns all environment variables for the
+  // authenticated user.
+  //
+  // +return NOT_FOUND User does not have access to a workspace with the given
+  // ID +return NOT_FOUND Workspace does not exist
+  rpc ListEnvironmentVariables(ListEnvironmentVariablesRequest)
+      returns (ListEnvironmentVariablesResponse) {}
+
+  // UpdateEnvironmentVariable updates an environment variable for the
+  // authenticated user.
+  //
+  // workspace_id +return NOT_FOUND Workspace does not exist
+  rpc UpdateEnvironmentVariable(UpdateEnvironmentVariableRequest)
+      returns (UpdateEnvironmentVariableResponse) {}
+
+  // DeleteEnvironmentVariable deletes an environment variable for the
+  // authenticated user.
+  //
+  // workspace_id +return NOT_FOUND Workspace does not exist
+  rpc DeleteEnvironmentVariable(DeleteEnvironmentVariableRequest)
+      returns (DeleteEnvironmentVariableResponse) {}
+}
+
+message UserEnvironmentVariable {
+  string id = 1;
+  string name = 2;
+  string value = 3;
+  string repository_pattern = 4;
+}
+
+message ListEnvironmentVariablesRequest { optional string workspace_id = 1; }
+
+message ListEnvironmentVariablesResponse {
+  repeated UserEnvironmentVariable environment_variables = 1;
+}
+
+message UpdateEnvironmentVariableRequest {
+  optional string envvar_name = 1;
```

Comment:
I think we should have here require `env_var_id`?

if env var belong to org then we should throw if envvar_repository_pattern is provided with invalid argument.